include/MemoryModel/PersistentPointsToCache.h

```diff
@@ -45,21 +45,27 @@ class PersistentPointsToCache
         initStats();
     }
 
+    /// Clear the cache.
+    void clear()
+    {
+      for (const Data *d : idToPts) delete d;
+      idToPts.clear();
+      ptsToId.clear();
+
+      unionCache.clear();
+      complementCache.clear();
+      intersectionCache.clear();
+    }
+
     /// Resets the cache removing everything except the emptyData it was initialised with.
     void reset(void)
     {
-        for (const Data *d : idToPts) delete d;
-        idToPts.clear();
-        ptsToId.clear();
+        clear();
 
         // Put the empty data back in.
         ptsToId[Data()] = emptyPointsToId();
         idToPts.push_back(new Data());
 
-        unionCache.clear();
-        complementCache.clear();
-        intersectionCache.clear();
-
```

Comment:
@mbarbar could you take a look at this part to make sure the refactoring is consistent as before?

Reply:
Looks good to me.